documents/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,947 @@
+your code to v2 from v1 of the AWS IoT SDK for Java.
+
+> [!NOTE]  
+> If you can't find the information you need in this guide, visit the [How to get help](#how-to-get-help) section for more help and guidance.
+
+* [What's new in AWS IoT Device SDK for Java v2](#whats-new-in-aws-iot-device-sdk-for-java-v2)
+* [How to get started with AWS IoT Device SDK for Java v2](#how-to-get-started-with-aws-iot-device-sdk-for-java-v2)
+    * [Package name change](#package-name-change)
+    * [MQTT protocol](#mqtt-protocol)
+    * [Client builder](#client-builder)
+    * [Connection types and features](#connection-types-and-features)
+    * [Lifecycle events](#lifecycle-events)
+    * [Publish](#publish)
+    * [Subscribe](#subscribe)
+    * [Unsubscribe](#unsubscribe)
+    * [Client stop](#client-stop)
+    * [Client shutdown](#client-shutdown)
+    * [Reconnects](#reconnects)
+    * [Offline operations queue](#offline-operations-queue)
+    * [Operation timeouts](#operation-timeouts)
+    * [Logging](#logging)
+    * [Client for AWS IoT Device Shadow](#client-for-aws-iot-device-shadow)
+    * [Client for AWS IoT Jobs](#client-for-aws-iot-jobs)
+    * [Client for AWS IoT fleet provisioning](#client-for-aws-iot-fleet-provisioning)
+    * [Example](#example)
+* [How to get help](#how-to-get-help)
+* [Appendix](#appendix)
+    * [MQTT5 features](#mqtt5-features)
+
+## What's new in AWS IoT Device SDK for Java v2
+
+* The v2 SDK client is truly async. Operations return `CompletableFuture` objects. Blocking calls can be emulated by waiting
+for the returned `CompletableFuture` object to be resolved.
+* The v2 SDK provides implementation for MQTT5 protocol, the next step in evolution of the MQTT protocol.
+* Public API terminology has changed. You `start()` or `stop()` the MQTT5 client rather than `Connect` or `Disconnect`
+as in the v1 SDK. This removes the semantic confusion between client-level controls and internal recurrent networking events related to connection and disconnection.
+* The v2 SDK supports AWS IoT services such as Jobs and fleet provisioning.
+
+Public API for almost all actions and operations has changed significantly. For more information about the new features
+and specific code examples, refer to the [How to get started with AWS IoT Device SDK for Java v2](#how-to-get-started-with-aws-iot-device-sdk-for-java-v2)
+section of this guide.
+
+## How to get started with AWS IoT Device SDK for Java v2
+
+There're differences between the v1 SDK and the v2 SDK. This section describes the changes you need to apply to your project
+with the v1 SDK to start using the v2 SDK. For more information about MQTT5, visit [MQTT5 User Guide](https://github.com/aws/aws-iot-device-sdk-java-v2/blob/main/documents/MQTT5_Userguide.md#getting-started-with-mqtt5).
+
+### Package name change
+
+A noticeable change from the v1 SDK to the v2 SDK is the package name change. Package names begin with `software.amazon.awssdk`
```

Comment:
Trivial: I don't know that we need the leading sentence to point out that a change is obvious. We could probably just state "Package names for the v2 SDK now begin with `software.amazon.awssdk` instead of `com.amazonaws` which was used by the v1 SDK."